### ISSUE TITLE
MixedClusterClientYamlTestSuiteIT times out more frequently lately

### ISSUE THREAD

**Issue by nik9000** [elastic/elasticsearch#69973]
**Build scan**: https://gradle-enterprise.elastic.co/s/4plpdtisgjwnu/

**Repro line**: Not going to be useful.

**Reproduces locally?**: No.

**Applicable branches**: 7.11, 7.12, 7.x. Happens on master, but only on windows as far as I can see.

**Failure history**:
![image](https://user-images.githubusercontent.com/215970/109985142-71411880-7cd2-11eb-9aba-e96b9a5d16f6.png)
https://build-stats.elastic.co/goto/2e7535c11ed73598b069d68b420f2014

**Failure excerpt**:
```
  2> WARNING: Suite execution timed out: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
  2> ==== jstack at approximately timeout time ====
  2> "I/O dispatcher 576" ID=625 RUNNABLE (in native code)
  2> 	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
  2> 	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
stack traces looked boring when I scanned them
```

**Comment by mark-vieira** [comment]
It seems it's possible the items related to #69641 didn't completely resolve this issue. We see more failures on Windows due to that system being slower already (not using a ramdisk) so any general regression in test execution times is going to plague those builds more. 

We can see this isn't specific to platform though. There's a clear regression in execution time for this test suite even on linux (below graph doesn't include Windows builds):

![image](https://user-images.githubusercontent.com/4106672/110020747-c6832700-7cde-11eb-8acb-324fd8fde0f7.png)

I believe the the fixes mentioned in #69641 should have percolated through CI already at this point. It sounds like there's still more work to do here to track down why these tests are taking so much longer.

**Comment by mark-vieira** [comment]
@jakelandis I hate to just tag Core/Features on this one again but I just want to ensure _someone_ has eyes on this. We have a larger problem of these very coarse tests suites not having clear ownership which I intend to bring up to a larger audience.

**Comment by mark-vieira** [comment]
I believe the build scan @nik9000 linked to is incorrect, Here's a recent timeout failure: https://gradle-enterprise.elastic.co/s/qwnmut63k53hq

**Comment by mark-vieira** [comment]
I think this is a larger more serious issue. Here is the trend of our BWC jobs on the `7.x` branch over the past month. Average build times have _doubled_ here. Keep in mind, these builds on are Linux using ramdisks. We've introduced some serious regression somewhere here.

![image](https://user-images.githubusercontent.com/4106672/110021954-3e058600-7ce0-11eb-8314-e8f8febd6642.png)

**Comment by jakelandis** [comment]
another example: https://gradle-enterprise.elastic.co/s/wdxaxxugsaka6

Both `:qa:mixed-cluster:v6.8.15#mixedClusterTest`, but different actual tests , and both on Linux. 

in one case:
MixedClusterClientYamlTestSuiteIT » test {p0=search/160_exists_query/Test exists query on object inner field in empty index}

and in the other:
MixedClusterClientYamlTestSuiteIT » test {p0=indices.get_alias/10_basic/Get aliases via /name,name/_alias/{name}}

Also no smoking gun like regex or long pauses in test (from a quick scan of the logs).

**Comment by mark-vieira** [comment]
Here's the build times for `master`. There's a slight increase around the same time period but not quite as extreme. I suspect that the BWC testing against 6.8.15 is more problematic.

![image](https://user-images.githubusercontent.com/4106672/110022577-ee738a00-7ce0-11eb-99e5-a7b872dafea4.png)

**Comment by nik9000** [comment]
> I believe the build scan @nik9000 linked to is incorrect, Here's a recent timeout failure: https://gradle-enterprise.elastic.co/s/qwnmut63k53hq

Looks like I linked to a regular yaml test timing out. I got my wires crossed. But its also bad that that task fails.

**Comment by mark-vieira** [comment]
Jake and I did some digging here and I think we've nailed down a timeframe where we thing some kind of regression snuck into `MixedClusterClientYamlTestSuiteIT`. He's going to follow up on doing further investigations here. Thanks, Jake!

**Comment by jakelandis** [comment]
OK ... pretty sure I got to the bottom of this. 

https://github.com/elastic/elasticsearch/pull/69153 changed the distribution used by this (and other tests) to use the default distribution. This results in a different code flow in `ESRestTestCase` when cleaning up after itself. There is a check in there `hasXpack` that changes how the clean up of the composable templates and performed. Prior to that change, it hit a work flow where the first `_index_template/*` would (correctly) silently error and move on. In the new workflow with the default distribution, the logic is different and it appears that there is more work being done. For example instead of `*` it is deleting in a loop, and instead of short circuiting the call to delete the composable templates via the original exception, a call is being made get the composable templates, erroring and producing a stack trace. 

All of this is very minimal per test... up to ~200ms slower...however, 200ms * 1300 test , then add on a additional full round of tests (due to a new staging branch) and that 40m window isn't quite good enough. 

To reproduce 

Use this command:
```
./gradlew  :qa:mixed-cluster:v6.8.15#mixedClusterTest  -Dtests.method="test {p0=search/220_total_hits_object/*}" --info
```
and this debug logging
```patch
diff --git a/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java b/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
index 799a21a95fb..7395b983ee5 100644
--- a/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
@@ -203,6 +203,7 @@ public abstract class ESRestTestCase extends ESTestCase {
                 nodeVersions.add(Version.fromString(nodeInfo.get("version").toString()));
                 for (Object module: (List<?>) nodeInfo.get("modules")) {
                     Map<?, ?> moduleInfo = (Map<?, ?>) module;
+                    logger.info("**** moduleInfo.get(\"name\"): " + moduleInfo.get("name"));
                     if (moduleInfo.get("name").toString().startsWith("x-pack-")) {
                         hasXPack = true;
                     }
@@ -593,8 +594,12 @@ public abstract class ESRestTestCase extends ESTestCase {
             wipeAllIndices();
         }
 
+        logger.info("**** Clearing preserveTemplatesUponCompletion [{}]", preserveTemplatesUponCompletion() );
         // wipe index templates
         if (preserveTemplatesUponCompletion() == false) {
+
+
+            logger.info("**** hasXpack [{}]", hasXPack() );
             if (hasXPack) {
                 /*
                  * Delete only templates that xpack doesn't automatically
@@ -605,6 +610,7 @@ public abstract class ESRestTestCase extends ESTestCase {
                 Request request = new Request("GET", "_cat/templates");
                 request.addParameter("h", "name");
                 String templates = EntityUtils.toString(adminClient().performRequest(request).getEntity());
+                logger.info("**** \"\".equals(templates)[{}]", "".equals(templates) );
                 if (false == "".equals(templates)) {
                     for (String template : templates.split("\n")) {
                         if (isXPackTemplate(template)) continue;
@@ -648,13 +654,16 @@ public abstract class ESRestTestCase extends ESTestCase {
                     // We hit a version of ES that doesn't support index templates v2 yet, so it's safe to ignore
                 }
             } else {
-                logger.debug("Clearing all templates");
+                logger.info("Clearing all templates");
                 adminClient().performRequest(new Request("DELETE", "_template/*"));
                 try {
+                    logger.info("***** before delete operation");
                     adminClient().performRequest(new Request("DELETE", "_index_template/*"));
                     adminClient().performRequest(new Request("DELETE", "_component_template/*"));
+                    logger.info("***** after delete operation");
                 } catch (ResponseException e) {
                     // We hit a version of ES that doesn't support index templates v2 yet, so it's safe to ignore
+                    logger.info("***** after delete operation - in catch");
                 }
             }
         }
```
Compare a current version (from today) 785a17c687ce9980ed60542467433bea0f38486e (slower) to something before  https://github.com/elastic/elasticsearch/pull/69153 such as 16a1662d372b114016b7ce23f2a1cb26a4b306c6 (faster) and you can see the exact difference. You can see my notes from testing here: https://gist.github.com/jakelandis/428f5f62070c51047db86cf950190e19

I am pretty sure if we can make the xPack workflow closer to the non-xPack workflow we will gain back those ~200ms and the tests will still fit in the 40m window.

**Comment by martijnvg** [comment]
Thanks for the analysis @jakelandis. Like you mention we should be able to reduce the time it takes to cleanup up all the templates:

- As a first change we can speedup the template cleanup logic itself by not attempting to first delete a template as legacy template and if that fails delete it as composable index template. We should get the list of all legacy templates and if exist delete those (filtering out xpack related ones) then get a list of all composable legacy templates and deletes those (minus the xpack templates).
- If this first change doesn't help to get running all the mixed cluster tests within the 40m time window then we can consider changing the legacy template delete api, composable index template delete api and component template delete api to accept multiple templates.
- Or alternatively change the names of the xpack templates, so that they have a shared prefix, which should allow the testing code to delete all the templates except the xpack templates. Although I'm not so sure what bwc consequences renaming the name of the templates will have.

**Comment by martijnvg** [comment]
There are still a number of test failures caused by test suite timeouts:
* https://gradle-enterprise.elastic.co/s/fytwocpn7skva
* https://gradle-enterprise.elastic.co/s/ekx6ca34eraa2
* https://gradle-enterprise.elastic.co/s/4mathbjux5orm

I think that when #70094 gets merged and backported and the same optimisation happens for the delete component api then this should hopefully fix all these test suite timeouts.

**Comment by davidkyle** [comment]
More instances of the suite timeouts today:
- https://gradle-enterprise.elastic.co/s/xdul3pmd7hqk6
- https://gradle-enterprise.elastic.co/s/5ichlrrmsxtna
- https://gradle-enterprise.elastic.co/s/txqma2kodkmsi

**Comment by martijnvg** [comment]
Changing the delete composable index template api (#70094) and the delete component template api (#70314) has helped reducing the test time for the `MixedClusterClientYamlTestSuiteIT` test suit, but only when running the mixed cluster test suite against new versions (7.13.0 and up). Almost all of the test suite timeouts were occurring when running the tests against 6.8.x clusters. For example on my machine running `./gradlew -p qa/mixed-cluster/ v6.8.15#mixedClusterTest` currently takes about 35 minutes (this used to be around 40 minutes which caused test suite time outs locally). While running the `mixedClusterTest` task against newer versions (7.5.0 or 7.11.3), it takes about 31 minutes. And running against a mixed 7.13.0-snapshot test cluster then the total build time is about 27 minutes. The improvement made via #70361 to base rest test class, helped reducing the number of test suite timeouts when running against mixed clusters with version 6.8.x.

I noticed just two mixed cluster suite test failure due to suite timeout being reached in the last 24 hours, which is a great improvement over what it used to be. However the fact that it still takes a long time to complete the `MixedClusterClientYamlTestSuiteIT` test suite against an individual test cluster makes me wonder whether the suite timeout for `MixedClusterClientYamlTestSuiteIT` should be increased? A random hick-up can easily cause this test suite to go over the configured 40 minutes limit (especially when running this test suite against a mixed cluster with older nodes).

**Comment by mark-vieira** [comment]
We can't just keep bumping the timeout. We've done this other places and it just becomes more and more problematic. Here's a graph of the execution time for this suite over the last few months (this is the default distro build, so includes that overhead vs oss):

![image](https://user-images.githubusercontent.com/4106672/111546543-1200f080-8735-11eb-9b81-247067c18542.png)

If we have single test classes that are predictably taking 30+ minutes to complete we need to consider chopping these things up.

**Comment by martijnvg** [comment]
> If we have single test classes that are predictably taking 30+ minutes to complete we need to consider chopping these things up.

You mean like splitting `MixedClusterClientYamlTestSuiteIT` up in multiple test suites? Currently `MixedClusterClientYamlTestSuiteIT` executes more than 1300 yaml tests. 

Several of the made improvements, were easy to spot in the server logs (for example using apis that were not supported or deleting builtin templates that immediately get re-added after removal). I will take another look at this. The server logs were very large, but now are much smaller (< 1MB). I suspect that the many deprecation warnings (type removal warnings and the default number of primary shards change) returned by 6.8.x cluster may be the cause why running against a 6.8.x cluster is slower than against a more recent version.

**Comment by mark-vieira** [comment]
> You mean like splitting MixedClusterClientYamlTestSuiteIT up in multiple test suites? Currently MixedClusterClientYamlTestSuiteIT executes more than 1300 yaml tests.

The even better thing would be to split the `:qa:mixed-cluster` project into multiple pieces. That way each of the divisions could run in parallel.

**Comment by nik9000** [comment]
We had support for hashing the test name across several tests years ago. I don't recall why we dropped it. Its that kind of thing @rjernst might remember something about. Presumably we dropped it for a good reason and if we add something like it back in we probably should remember why we removed it the first time around.

**Comment by mark-vieira** [comment]
> We had support for hashing the test name across several tests years ago. I don't recall why we dropped it. Its that kind of thing @rjernst might remember something about. Presumably we dropped it for a good reason and if we add something like it back in we probably should remember why we removed it the first time around.

Can you elaborate. I'm not sure what you mean by "hashing the test name".

**Comment by nik9000** [comment]
> Can you elaborate. I'm not sure what you mean by "hashing the test name".

We broke the test suite up arbitrarily so we could parallelize it. We had a suite that ran all tests with names like `testName.hashcode() % 4 == 0` and another one with `testName.hashcode() % 4 == 1` and another one with `== 2`, etc. Then we ran those in parallel.

**Comment by mark-vieira** [comment]
Cool. Yeah I don't recall that ever being a thing for any kind of integration tests. They were always serialized so far as I knew as we don't have strong guarantees on whether it's "ok" to run them in parallel.

If parallel execution of YAML test suites is something that should be supported then I'm 100% for doing this.

**Comment by nik9000** [comment]
> If parallel execution of YAML test suites is something that should be supported then I'm 100% for doing this.

The parallel execution was a long, long time ago. I'm fairly sure we'd need totally separate clusters to do any parallelization. And it'd only work on the "stateless" yaml tests. But the ones in mixed cluster tests and `rest-api-spec` are stateless. And I'm sure there are other problems with splitting up the tests because computers.

**Comment by mark-vieira** [comment]
Yeah, that's why the "easiest" thing here would be to cut up the current suite into multiple subprojects. Each would use it's own test cluster. For example, right now this runs the entire core test suite, serially, against a single cluster. We could cut this up pretty easily I think.

**Comment by martijnvg** [comment]
Splitting up the current test suite into multiple sub-projects sounds like a good idea. I'm curious how this would like. How are the tests going to be split? Are we going to have a subproject per bwc version and some subset of tests?

**Comment by mark-vieira** [comment]
My initial thought was subproject per some subset of tests. No need to bust up by bwc version, that would get too complicated I think.

So instead of doing `includeCore '*'` when bringing in yaml tests we'd explicitly list the suites we want for that particular project.

**Comment by nik9000** [comment]
> So instead of doing `includeCore '*'` when bringing in yaml tests we'd explicitly list the suites we want for that particular project.

I think it's important that we have a "catch all" if we do that.

**Comment by mark-vieira** [comment]
> I think it's important that we have a "catch all" if we do that.

Agreed. It might make sense to add an "exclude", such that we include all, then exclude the ones expressly configured in the othe r subprojects.

**Comment by rjernst** [comment]
> We had support for hashing the test name across several tests years ago.

I don't recall when exactly we stopped doing this, but I do vaguely remember the problems. As noted in the PR (https://github.com/elastic/elasticsearch/pull/21391) in which I removed the underlying support for doing this (after we had already stopped running those style tests), I noted reproducibility as one issue. I believe the problem was when we ran a reproduce line, we would not get the same set of tests to run before it. At the time, we were still running a single cluster, but multiple parallel test runners. So if there were any side effects from a test, it could cause another test to fail in inexplicable ways.

With distinct subprojects we would have separate test clusters and runners. This sounds good to me. The only wish I have is that it could somehow automatically balance the tests. That is, we have N clusters which are subprojects, but each of those pull the tests to run based on hashing of the suite name. Otherwise I'm afraid manual allocation will get imbalanced quickly as new suites are added.

**Comment by nik9000** [comment]
>  The only wish I have is that it could somehow automatically balance the tests.

That sounds like a work queue style thing. Something like that might be easier to do if we don't break into sub-projects and instead spin up a couple of clusters in the same project and limit the runner to n parallel tasks. Or something like that.

**Comment by rjernst** [comment]
I don't think it has to be a work queue. Separate projects can still have the rest test suites assigned statically. My concern is about *manual* assignment, which can easily get unbalanced, especially with a catch-all as you described which would effectively pickup new tests.

**Comment by nik9000** [comment]
Fair enough!

I think your point about reproduction lines is important too. It'd be sad if moving the test from one subproject to the other broke reproduction.

**Comment by mark-vieira** [comment]
> My concern is about manual assignment, which can easily get unbalanced

This will never be perfect, we have to continually balance things. We run into the same issue in CI where some jobs become much longer than others and we need to revisit. Ideally yes, we would auto balance and spread load across any number of test workers but we aren't there yet. An unbalanced set is still much better than having only a single test runner.

**Comment by ywelsch** [comment]
Another timeout: https://gradle-enterprise.elastic.co/s/ezacnhjkbru7m

**Comment by martijnvg** [comment]
All test timeouts seem to occur when running the `:qa:mixed-cluster:v6.8.x#mixedClusterTest` gradle project. I don't recall seeing timeouts occur when running mixed cluster tests against a 7.x test cluster. I'm not sure what makes the test suite execute slower against a 6.8.x test cluster. I do remember seeing that many deprecation warnings are being omitted (type removal, default number of primary shards change from 5 to 1). Maybe that has something to do with the test suite timeouts against a 6.8.x test cluster.

**Comment by hendrikmuhs** [comment]
Another one: https://gradle-enterprise.elastic.co/s/dsq62q3ajsdf6

(FWIW 6.8 again)

**Comment by valeriy42** [comment]
Another one: https://gradle-enterprise.elastic.co/s/xkl4k3ykydlza

**Comment by BigPandaToo** [comment]
7.13 and 7.x again:
https://groups.google.com/a/elastic.co/g/build-elasticsearch/c/xB-jlAiHMks

**Comment by davidkyle** [comment]
> All test timeouts seem to occur when running the :qa:mixed-cluster:v6.8.x#mixedClusterTest gradle project.

Most are against v6.8.16 but there are exceptions. Here's one on master for `:qa:mixed-cluster:v7.14.0#mixedClusterTest` https://gradle-enterprise.elastic.co/s/im2vobplu4l4u

And `':qa:mixed-cluster:v7.12.1#mixedClusterTest'` on the 7.x branch https://gradle-enterprise.elastic.co/s/5ggdnip75vi7y

**Comment by mark-vieira** [comment]
Is there anything else we can do here? The rest-api-spec tests (the exact same YAML tests being executed here) run on Windows in 18 minutes. The mixed cluster tests however timeout consistently after 30 minutes. What exactly is so costly in the mixed-cluster scenario? This is effectively failing our Windows builds 100% of the time, not to mention these tests are just insanely expensive to run compared to other test suites. Given that it's probably worth taking another look at what we can do here.

**Comment by mark-vieira** [comment]
The mixed-cluster tests against 6.8 take 50% longer while running only 60% the number of tests. Is having a 6.8 node in the cluster just slowing everything down or is it some kind of handling in our test code?

![image](https://user-images.githubusercontent.com/4106672/116276079-02fc5d80-a739-11eb-9739-505db7e3e5aa.png)

**Comment by benwtrent** [comment]
another failure: https://gradle-enterprise.elastic.co/s/6r7eu5ti4yinm

Unsurprisingly, its a timeout with a 6.8 node.

**Comment by romseygeek** [comment]
Another failure, on Windows, with a 6.8 node: https://gradle-enterprise.elastic.co/s/npap2fcznlsfe

**Comment by ywangd** [comment]
Another one: https://gradle-enterprise.elastic.co/s/gccurjqc5zeuc
Same timeout, different method `MixedClusterClientYamlTestSuiteIT » test {p0=indices.stats/12_level/Level - blank}`

It's a 7.14 with a 6.8.19 cluster.

**Comment by Luegg** [comment]
We got another one today https://gradle-enterprise.elastic.co/s/v7cnbtghenhew This time on `MAC` and 7.15.

**Comment by valeriy42** [comment]
Another one today https://gradle-enterprise.elastic.co/s/iaqp4ej5iaxmk . 

In the past 7 days, there were [21 time-outs](https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=Europe/Berlin&tests.container=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT&tests.sortField=FAILED&tests.test=classMethod&tests.unstableOnly=true). Only `MAC` and `WINDOWS` were affected, `7.x`, `7.14` and `7.15`.

**Comment by DaveCTurner** [comment]
Another at https://gradle-enterprise.elastic.co/s/zzef53eape25s

**Comment by ywelsch** [comment]
Another at https://gradle-enterprise.elastic.co/s/mp4z4klsb2h5w

**Comment by mark-vieira** [comment]
I'm really not sure how to proceed on this issue. The fact of the matter is this test suite just seems impossibly slow when running a mixed cluster on `7.x` with a `6.8` node. On Windows (where _everything_ is slower) this is simply causing things to time out. Again, I'm not sure who "owns" this test suite given it's cross functional but clearly something is funny with this configuration. Can anyone explain to me why a cluster with a `6.8` node and `7.x` node would perform so poorly? Is this simply to be expected or is there something wrong with the test setup?

That being said, even the non-6.8 scenarios take 30 minutes to execute on Windows (roughly 50% longer than on Linux). Even in the "good" scenarios though, Linux, and even using a ramdisk, this test suite is insanely expensive to run, taking over 15 minutes in the best case scenario for each BWC version. Essentially, this is just a exacerbated version of the `:rest-api-spec` test suite which also continues to grow, and grow, and grow. We need a way to chop this up into smaller chunks, and ideally run more in parallel.

**Comment by mark-vieira** [comment]
I've opened https://github.com/elastic/elasticsearch/issues/78043 to more generally address the root problem here, which is that we have monolith test suites that just grow and grow and grow.

**Comment by fcofdez** [comment]
Another at https://gradle-enterprise.elastic.co/s/mkqdcqpdittho

And I've seen timeouts in `ClientYamlTestSuiteIT` and `DocsClientYamlTestSuiteIT` too

**Comment by gwbrown** [comment]
Another one on [7.15 on windows](https://gradle-enterprise.elastic.co/s/ylviumab2tdqm/tests/:qa:mixed-cluster:v6.8.21%23mixedClusterTest/org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT/test%20%7Bp0=indices.delete%2F10_basic%2FDelete%20index%20against%20wildcard%20matching%20alias%20-%20disallow%20no%20indices%20-%20multiple%20indices%7D?top-execution=1), and another on [8.0 on Darwin](https://gradle-enterprise.elastic.co/s/amfrmzy5yakuw/tests/:docs:integTest/org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT/test%20%7Byaml=reference%2Fanalysis%2Ftokenfilters%2Fdecimal-digit-tokenfilter%2Fline_75%7D?top-execution=1).

**Comment by jkakavas** [comment]
Another one on 7.16 on Darwin https://gradle-enterprise.elastic.co/s/nciwkqhxbjv2o/tests/:qa:mixed-cluster:v7.16.4%23mixedClusterTest/org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT/test%20%7Bp0=cat.indices%2F10_basic%2FTest%20cat%20indices%20sort%7D?top-execution=1

**Comment by martijnvg** [comment]
Two more instances:

- https://gradle-enterprise.elastic.co/s/rbkgssegs7eva (7.17 Darwin)
- https://gradle-enterprise.elastic.co/s/txhps7ujw4phg (7.17 Darwin)

**Comment by valeriy42** [comment]
Another instance https://gradle-enterprise.elastic.co/s/cjr2nsx6pi5ae (7.17 Linux)

**Comment by andreidan** [comment]
Another one 7.17 Linux https://gradle-enterprise.elastic.co/s/5laehinvpzpty

**Comment by mark-vieira** [comment]
> Another one 7.17 Linux https://gradle-enterprise.elastic.co/s/5laehinvpzpty

I think the 7.17 failure is just a temporary incident. Looking at execution time for this suite over the past week it's been quite stead and well below the 40m timeout threshold so I don't think this commonly happens on that branch.

![image](https://github.com/elastic/elasticsearch/assets/4106672/c5023dd9-6960-4041-8cf6-2108df2f8e8d)

**Comment by mark-vieira** [comment]
Categorizing this as low-risk since we haven't muted anything. There were 14 instances of this suite timing out in the past week. That majority were in release, concurrent search, and encryption at rest tests. These all have in common that the run the entire test suite in one big giant build meaning lots of resource contention which contributes to these timeouts. I'm wondering if we should do the same as we did for the platform support jobs and skip BWC testing in these jobs.

cc @brianseeders

**Comment by mark-vieira** [comment]
So the majority of these timeouts continue to be release and encryption at rest tests. I think we should probably due the following:

1. For release tests, either split it up into parts, or run them in-memory, or both. There's no reason to run the entire suite, on a single machine, on disk.
2. For the EAR tests I think we can just disable BWC. It's probably worthwhile to run BWC tests in release mode, but I see little to no benefit to running these tests in the EAR job.

**Comment by mark-vieira** [comment]
Closing this for now. If it continues to happen test triage will reopen.
